src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.cs

```diff
@@ -12,6 +12,11 @@ namespace System.Diagnostics
     /// </summary>
     public partial class DebugProvider
     {
+        private readonly bool _ignoreIndentation;
+
+        public DebugProvider() { _ignoreIndentation = false; }
+        public DebugProvider(bool ignoreIndentation) { _ignoreIndentation = ignoreIndentation; }
```

Comment:
After Debug gets wired with TraceListeners, then Indentation gets done through the Trace Listeners instead.
Without `_ignoreIndentation` flag, we would get double indentation because both the trace listener and debug provider will apply the indentation.

TODO: Add comment to this constructor explaining why it's useful